pyrk/th_system.py

```diff
@@ -80,27 +247,100 @@ def convection(self, t_b, t_env, h, A):
+        cap = (component.rho(t_idx)*component.cp*component.vol)
+        if component.heatgen:
+            to_ret += self.heatgen(component, power, omegas)/cap
+        for name, d in component.adv.iteritems():
+            Qadv = self.advection(t_out=component.T[t_idx]*2.0 - d['t_in'],
+                                  t_in=d['t_in'],
+                                  m_flow=d['m_flow'],
+                                  cp=d['cp'])
+            to_ret -= Qadv/cap
+            assert Qadv.magnitude >= 0, '''
+            at step %d, %s is heating the system by %f watts???
+            Tin is %f, Tout is %f, Tcomp is %f''' % (
+                t_idx, component.name, Qadv.magnitude, d['t_in'].magnitude,
+                (component.T[t_idx]*2 - d['t_in']).magnitude,
+                component.T[t_idx].magnitude)
+        for interface, d in component.cond.iteritems():
+            env = self.comp_from_name(interface)
+            Qcond = self.conduction(t_b=component.T[t_idx],
+                                    t_env=env.T[t_idx],
+                                    r_b=d['r_b'],
+                                    r_env=d['r_env'],
+                                    k=d['k'])
+            to_ret -= Qcond/cap
+            assert (Qcond*(component.T[t_idx]-env.T[t_idx])).magnitude >= 0, \
+                'conduction from %s to %s, from temp %f to %f is wrong %f' % (
+                    component.name, env.name, component.T[t_idx].magnitude,
+                    env.T[t_idx].magnitude, Qcond.magnitude)
+        for interface, d in component.conv.iteritems():
+            env = self.comp_from_name(interface)
+            Qconv = self.convection(t_b=component.T[t_idx],
+                                    t_env=env.T[t_idx],
+                                    h=d['h'],
+                                    A=d['area'])
+            to_ret -= Qconv/cap
+            assert (Qconv*(component.T[t_idx]-env.T[t_idx])).magnitude >= 0,\
+                'convection from %s to %s, from temp %f to %f is wrong %f' % (
+                    component.name, env.name, component.T[t_idx].magnitude,
+                    env.T[t_idx].magnitude, Qconv.magnitude)
+        return to_ret
+
+    def conduction(self, t_b, t_env, r_b, r_env, k):
         """
+        heat transfer by conduction(watts)
         :param t_b: The temperature of the body
         :type t_b: float.
         :param t_env: The temperature of the environment
         :type t_env: float.
-        :param L: the length scale of the body
-        :type L: float.
+        :param r_b: The radius of body
+        :type r_b: float.
+        :param r_env: The radius of environment
+        :type r_env: float.
         :param k: the thermal conductivity
         :type k: float.
-        :param A: the surface area of heat transfer
-        :type A: float.
         """
-        num = (t_b-t_env)
-        denom = L/(k*A)
+        num = 4*math.pi*k*(t_b-t_env)
+        denom = (1/r_b - 1/r_env)
```

Comment:
I haven't worked on the ThSystemSphPS class. I am not sure if we need to
keep it. It‘s not a realistic model.

On Sat, Oct 3, 2015 at 5:47 PM, Katy Huff <notifications@github.com> wrote:

> In pyrk/th_system.py
> <https://github.com/pyrk/pyrk/pull/10#discussion_r41093198>:
>
> >          """
> > -        num = (t_b-t_env)
> > -        denom = L/(k*A)
> > +        num = 4*math.pi*k*(t_b-t_env)
> > +        denom = (1/r_b - 1/r_env)
>
> lines 326 and 327 make all conduction spherical conduction. This will not
> be appropriate for non-spherical lumps. This type of conduction should only
> be called if the lump is a sphere... right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyrk/pyrk/pull/10/files#r41093198>.
>



-- 
-----------------------------------------
Xin WANG | 王昕
PhD Student
Thermal Hydraulics Laboratory
Nuclear Engineering Department
University of California, Berkeley
(415)408-1691